jctools-core/src/main/java/org/jctools/util/PaddedAtomicLong.java

```diff
@@ -0,0 +1,301 @@
+package org.jctools.util;
+
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.function.LongBinaryOperator;
+import java.util.function.LongUnaryOperator;
+
+import static org.jctools.util.UnsafeAccess.UNSAFE;
+import static org.jctools.util.UnsafeAccess.fieldOffset;
+
+abstract class PaddedAtomicLongSuper extends Number implements java.io.Serializable {
+    private static final long serialVersionUID = 1;
+
+    transient long p01, p02, p03, p04, p05, p06, p07;
+    transient long p10, p11, p12, p13, p14, p15, p16, p17;
+}
+
+abstract class PaddedAtomicLongFields extends PaddedAtomicLongSuper {
+    volatile long value;
+}
+
+abstract class PaddedAtomicLongSub extends PaddedAtomicLongFields {
+
+    transient long p01, p02, p03, p04, p05, p06, p07;
+    transient long p10, p11, p12, p13, p14, p15, p16, p17;
+}
+
+/**
+ * A padded version of the {@link java.util.concurrent.atomic.AtomicLong}.
+ */
+public class PaddedAtomicLong extends PaddedAtomicLongSub {
```

Comment:
By convention we use Pad0,Pad1 etc for the padding classes rather than Super/Sub

Reply:
Fixed